quiz/src/Category.js

```diff
@@ -0,0 +1,35 @@
+import React ,{useState,useEffect} from 'react';
+import {CategoryData,DifficultyData} from './Data';
+
+function Category () {
+  const[cat,setCat]=useState('');
+  const[diff,setDiff]=useState('');
+  const [questions,setQuestions]=useState([]);
+  useEffect(()=>{
+    fetch(`https://opentdb.com/api.php?amount=10&category=${cat}&difficulty=${diff}`)
+    .then(res => res.json()).then(({results})=>setQuestions(results))
+  })
+
+  return(
+    <>
+    <form className='form-container'>
+
+     <label htmlFor='category'>Select category: </label>
+     <select name = {cat} onChange={(e)=> setCat(e.target.value)}>
+     {CategoryData.map((cat) => <option key={cat.id} value={cat.id}>{cat.name}</option>)}
+     </select>
```

Comment:
**Here, you used `htmlFor` with `label` to specifie which form element a label is bound to.**
so, 
you should use the attribute `id` with the desired element,  which here is `select` then give the `id` the same value of `htmlFor`